apps/web/src/components/Table/ResponsiveTable/useRowSelection.ts

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Updater, RowSelectionState, OnChangeFn } from "@tanstack/react-table";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+
+import { RowSelectDef } from "./types";
+
+type UseRowSelectionReturn = [
+  value: RowSelectionState,
+  setter: OnChangeFn<RowSelectionState>,
+];
+
+const useRowSelection = <T>(
+  data: T[],
+  rowSelect?: RowSelectDef<T>,
+): UseRowSelectionReturn => {
+  const [rowSelection, setRowSelection] = React.useState<RowSelectionState>({});
+
+  const rowSelectionCallback = (newRowSelection: RowSelectionState) => {
```

Comment:
Maybe the definitions of `rowSelectionCallback`, `handleRowSelection` and `setter` should use `useCallback`, to ensure the setter function returned by `useRowSelection` is as stable as possible?

Reply:
Heh, somehow you were viewing an old file. This was already done 😅 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/5f38b168475ee5ad4c7611cb4f3d9ecb08047ae7/apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx#L333-L334